### PR TITLE
Topic/display url review

### PR DESCRIPTION
**Pull request by i386**
# Description

Builds on top of #503.
* We needed to split the provider impl into its own plugin (did not belong to rest or config)
* display-url-provider 0.5 simplified calculating the base URL
* Never returns null
* Since multiple modules depend on pipeline and SCM APIs their maven module versions are defined in the root POM, as is maven best practice.
* Bumped github branch source to 2.10 as it has support for display url provider 💅 

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
So the aim of this is to provide URLs for notifications that support it to link to blue ocean right? 

I have a few problems with this approach: 

1) its a big big change for a small but important thing
2) it shouldn't be on by default, ie people installing blue ocean shouldn't have URLs changed just yet (maybe via a switch?)
3) it probably belongs in blueocean-pipeline-api-impl - if it has a dependency on pipeline
4) could it be deployed separately? 

Modules are like puppies, modules that are deployed as plugins, doubly so.

File: blueocean/pom.xml [hunk 1]
```diff
@@ -9,7 +9,7 @@
     </parent>
 
     <artifactId>blueocean</artifactId>
-    <packaging>hpi</packaging>
+    <packaging>pom</packaging>
```
Comment (michaelneale) [inline blueocean/pom.xml]: ???

Comment (i386) [inline blueocean/pom.xml]: oh thanks IDEA

Comment (michaelneale) [inline blueocean/pom.xml]: Thanks Obama

File: blueocean-testing/pom.xml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (michaelneale) [inline blueocean-testing/pom.xml]: IF we really want a discrete testing utils module - probably best done separately as a PR

File: blueocean-displayurl-provider/pom.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          <scope>test</scope>
+      </dependency>
+      <dependency>
+          <groupId>org.jenkins-ci.plugins</groupId>
+          <artifactId>git</artifactId>
+          <scope>test</scope>
+      </dependency>
+      <dependency>
+          <groupId>${project.groupId}</groupId>
+          <artifactId>blueocean-testing</artifactId>
```
Comment (michaelneale) [inline blueocean-displayurl-provider/pom.xml]: shouldn't this be test scope?

File: blueocean-pipeline-api-impl/pom.xml [hunk 4]
```diff
@@ -19,95 +19,86 @@
             <groupId>${project.groupId}</groupId>
             <artifactId>blueocean-rest-impl</artifactId>
         </dependency>
+
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>blueocean-testing</artifactId>
```
Comment (michaelneale) [inline blueocean-pipeline-api-impl/pom.xml]: also - scope?

File: blueocean-pipeline-api-impl/pom.xml [hunk 5]
```diff
@@ -19,95 +19,86 @@
...
+        </dependency>
+
         <dependency>
             <groupId>org.jenkins-ci.plugins.workflow</groupId>
             <artifactId>workflow-job</artifactId>
-            <version>2.6</version>
```
Comment (michaelneale) [inline blueocean-pipeline-api-impl/pom.xml]: any reason why these are being dropped as part of this PR?

**Comment by michaelneale** [comment]
lets park this until ivan is back. I think this should be in another module (and these other refactorings done separately)

**Comment by michaelneale** [comment]
@imeredith can we close this? (given you are working on it another way)

**Comment by michaelneale** [comment]
@imeredith we can chop this now right?

**Comment by michaelneale** [comment]
closing as not needed

**Pull request closed without merging**